sql/core/src/test/scala/org/apache/spark/sql/FileBasedDataSourceSuite.scala

```diff
@@ -842,6 +842,162 @@ class FileBasedDataSourceSuite extends QueryTest
       }
     }
   }
+
+  test("SPARK-31116: Select simple parquet columns correctly in case insensitive manner") {
```

Comment:
Do you target to test this in ORC too? Nested ORC column pruning was added as of SPARK-27034 in Spark 3.0. If you don't target to test ORC too, we might have to move the test case to Parquet specific suite.

Reply:
Oh, I just read Dongjoon's comment at https://github.com/apache/spark/pull/27888#discussion_r391491836. Okay, good to keep the tests here.